modules/account-lib/src/coin/stx/utils.ts

```diff
@@ -0,0 +1,156 @@
+  if (tx.auth.spendingCondition !== null && tx.auth.spendingCondition !== undefined) {
+    const spendingCondition = tx.auth.spendingCondition;
+    const txSender = getAddressFromPublicKeyHash(
+      Buffer.from(spendingCondition.signer, 'hex'),
+      spendingCondition.hashMode as number,
+      tx.version,
+    );
+    return txSender;
+  } else throw new Error('spendingCondition should not be null');
+}
+
+/**
+ * Returns whether or not the string is a valid amount number
+ *
+ * @param {string} amount - the string to validate
+ * @returns {boolean} - the validation result
+ */
+export function isValidAmount(amount: string): boolean {
+  const bigNumberAmount = new BigNumber(amount);
+  return bigNumberAmount.isInteger() && bigNumberAmount.isGreaterThanOrEqualTo(0);
+}
+
+/**
+ * Returns whether or not the string is a valid protocol address
+ *
+ * @param {string} address - the address to be validated
+ * @returns {boolean} - the validation result
+ */
+export function isValidAddress(address: string): boolean {
+  return validateStacksAddress(address);
+}
+
+/**
+ * Returns whether or not the string is a valid protocol transaction id or not.
+ *
+ * A valid transaction id is a SHA-512/256 hash of a serialized transaction; see
+ * the txidFromData function in @stacks/transaction:
+ * https://github.com/blockstack/stacks.js/blob/master/packages/transactions/src/utils.ts#L97
+ *
+ * @param {string} txId - the transaction id to be validated
+ * @returns {boolean} - the validation result
+ */
+export function isValidTransactionId(txId: string): boolean {
+  if (txId.length !== 64 && txId.length !== 66) return false;
+  const noPrefix = removeHexPrefix(txId);
+  if (noPrefix.length !== 64) return false;
+
+  return allHexChars(noPrefix);
+}
+
+/**
+ * Returns whether or not the string is a valid protocol public key
+ *
+ * The key format is documented at
+ * https://github.com/stacksgov/sips/blob/main/sips/sip-005/sip-005-blocks-and-transactions.md#transaction-authorization
+ *
+ * @param {string} pub - the  public key to be validated
+ * @returns {boolean} - the validation result
+ */
+export function isValidPublicKey(pub: string): boolean {
+  if (pub.length !== 66 && pub.length !== 130) return false;
+
+  const firstByte = pub.slice(0, 2);
+
+  // uncompressed public key
+  if (pub.length === 130 && firstByte === '04') return allHexChars(pub);
+
+  // compressed public key
+  if (firstByte === '02' || firstByte === '03') return allHexChars(pub);
+
+  return false;
+}
+
+/**
+ * Returns whether or not the string is a valid protocol private key
+ *
+ * The key format is described in the @stacks/transactions npm package, in the
+ * createStacksPrivateKey function:
+ * https://github.com/blockstack/stacks.js/blob/master/packages/transactions/src/keys.ts#L125
+ *
+ * @param {string} prv - the  private key to be validated
+ * @returns {boolean} - the validation result
+ */
+export function isValidPrivateKey(prv: string): boolean {
```

Comment:
I read the code linked above, but it kind of makes no sense to have a compressed private key - I'm not even sure what that means in the context of the secp256k1 curve (usually public keys are compressed, but not private keys). I'll assume that this is some strange implementation detail